src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -659,8 +659,14 @@ protected static Model createEffectivePom( ModelBuildingRequest buildingRequest,
                 public void injectProfile( Model model, Profile profile, ModelBuildingRequest request,
                                            ModelProblemCollector problems )
                 {
-
-                    // do nothing
+                    // Apply properties from profiles that are active by default. They might be needed to resolve
+                    // my own version.
+                    if (profile.getActivation() != null && profile.getActivation().isActiveByDefault()) {
+                        Properties merged = new Properties();
+                        merged.putAll(profile.getProperties());
+                        merged.putAll(model.getProperties());
```

Comment:
I think, the properties from the model need to be added to `merged` properties before the properties from the profile. Properties defined in a profile overwrite the ones in the model.